python/cudf/cudf/core/dataframe.py

```diff
@@ -3247,6 +3247,99 @@ def drop_duplicates(
 
         return self._mimic_inplace(outdf, inplace=inplace)
 
+    @_cudf_nvtx_annotate
+    def duplicated(self, subset=None, keep="first"):
+        """
+        Return boolean Series denoting duplicate rows.
+
+        Considering certain columns is optional.
+
+        Parameters
+        ----------
+        subset : column label or sequence of labels, optional
+            Only consider certain columns for identifying duplicates, by
+            default use all of the columns.
+        keep : {'first', 'last', False}, default 'first'
+            Determines which duplicates (if any) to mark.
+
+            - ``first`` : Mark duplicates as ``True`` except for the first occurrence.
+            - ``last`` : Mark duplicates as ``True`` except for the last occurrence.
+            - False : Mark all duplicates as ``True``.
+
+        Returns
+        -------
+        Series
+            Boolean series indicating duplicated rows.
+
+        See Also
+        --------
+        Index.duplicated : Equivalent method on index.
+        Series.duplicated : Equivalent method on Series.
+        Series.drop_duplicates : Remove duplicate values from Series.
+        DataFrame.drop_duplicates : Remove duplicate values from DataFrame.
+
+        Examples
+        --------
+        Consider a dataset containing ramen product ratings.
+
+        >>> import cudf
+        >>> df = cudf.DataFrame({
+        ...     'brand': ['Yum Yum', 'Yum Yum', 'Indomie', 'Indomie', 'Indomie'],
+        ...     'style': ['cup', 'cup', 'cup', 'pack', 'pack'],
+        ...     'rating': [4, 4, 3.5, 15, 5]
+        ... })
+        >>> df
+             brand style  rating
+        0  Yum Yum   cup     4.0
+        1  Yum Yum   cup     4.0
+        2  Indomie   cup     3.5
+        3  Indomie  pack    15.0
+        4  Indomie  pack     5.0
+
+        By default, for each set of duplicated values, the first occurrence
+        is set to False and all others to True.
+
+        >>> df.duplicated()
+        0    False
+        1     True
+        2    False
+        3    False
+        4    False
+        dtype: bool
+
+        By using 'last', the last occurrence of each set of duplicated values
+        is set to False and all others to True.
+
+        >>> df.duplicated(keep='last')
+        0     True
+        1    False
+        2    False
+        3    False
+        4    False
+        dtype: bool
+
+        By setting ``keep`` to False, all duplicates are True.
+
+        >>> df.duplicated(keep=False)
+        0     True
+        1     True
+        2    False
+        3    False
+        4    False
+        dtype: bool
+
+        To find duplicates on specific column(s), use ``subset``.
+
+        >>> df.duplicated(subset=['brand'])
+        0    False
+        1     True
+        2    False
+        3     True
+        4     True
+        dtype: bool
+        """  # noqa: E501
+        return super().duplicated(subset=subset, keep=keep)
```

Comment:
Looks like this method exists only so we have a distinct docstring. Is there any way to write a single docstring that encapsulates all `Series`, `DataFrame` and `Index` use-cases? If that feels like it would be doing too much, I'm fine with this.

Reply:
The docstrings of `duplicated` should now be `de-duplicated`.

Reply:
Sorry, I should have been more clear before. My question was more along the lines of, can we just delete this method altogether? All it does is call `super().method()`, and the only reason it exists is to override the docstring of the base class. Thus, can we write a single, generalized, base class docstring that all subclasses simply inherit? For example,

```python
class Frame:

    def duplicated():
        """
        Indicate duplicated rows of the input Series, Index, or DataFrame
        """
```

But, I realize that this is made difficult by the fact that `Series.duplicated` and `DataFrame.duplicated` take different positional arguments.

I'm not convinced the new approach using `_doc_duplicated_docstring` is better than what we were doing before my review comment (I'd rather live with the small amount of duplication than the added indirection and complexity).

Reply:
got it, could you see if the current changes that I force-pushed are satisfactory?

Reply:
Yup - looks great to me!